QModManager/API/SMLHelper/Crafting/ModCraftTreeFamily.cs

```diff
@@ -0,0 +1,446 @@
+﻿namespace QModManager.API.SMLHelper.Crafting
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using Assets;
+    using Patchers;
+    using QModManager.Utility;
+    using UnityEngine;
+    using UnityEngine.Assertions;
+    using Utility;
+
+    /// <summary>
+    /// Basic data structure of a crafting tree node.
+    /// </summary>
+    public abstract class ModCraftTreeNode
```

Comment:
@AlexejheroYTB If you agree, we can create a new folder `QModManager/API/SMLHelper/Crafting/CraftTreeModding` and split all these classes into their own separate files.

Reply:
Yea, the file is very cluttered and contains a lot of classes. I'll let you split them up as you want. You can even change the namespaces as we don't need this to be backwards-compatible.

Reply:
Refactor cannot be done without first moving the SMLHelper shared project outside of the QModManager project folder.